exercises/src/test/scala/state/StateSpec.scala

```diff
@@ -0,0 +1,200 @@
+import org.specs2.mutable.Specification
+import org.specs2.ScalaCheck
+
+import fpinscala.state._
+
+class StateSpec extends Specification with ScalaCheck {
+ 
+ //========================================================================================
+ //#6.1
+  "nonNegativeInt" should {
+    "return a bigger than or equal to zero Int Output along with a new State" in {
+      prop { 
+        x: Int =>
+          val rng1 = new RNG.Simple(x)
```

Comment:
Not a big deal, but no need for `new` when creating an instance of a case class.